src/SourceBuild/content/Directory.Build.targets

```diff
@@ -2,17 +2,24 @@
 
   <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" Condition="'$(SkipArcadeSdkImport)' != 'true'" />
 
-  <Target Name="DetermineMicrosoftSourceBuildIntermediateInstallerVersion"
-          Condition="'$(DotNetBuildFromSource)' == 'true'">
-    <!-- Manually load the installer version from the PVP. -->
-    <XmlPeek XmlInputPath="$(CollapsedIntermediatePath)PackageVersions.package-source-build.Current.props"
-             Query="msb:Project/msb:PropertyGroup/msb:MicrosoftSourceBuildIntermediateInstallerVersion/text()"
-             Namespaces="&lt;Namespace Prefix='msb' Uri='http://schemas.microsoft.com/developer/msbuild/2003'/&gt;">
-        <Output TaskParameter="Result" ItemName="MicrosoftSourceBuildIntermediateInstallerVersionItem" />
-    </XmlPeek>
+  <Target Name="DetermineSourceBuiltSdkVersion">
     <PropertyGroup>
-      <MicrosoftSourceBuildIntermediateInstallerVersion>@(MicrosoftSourceBuildIntermediateInstallerVersionItem)</MicrosoftSourceBuildIntermediateInstallerVersion>
-      <MicrosoftSourceBuildIntermediateInstallerVersion Condition="'$(MicrosoftSourceBuildIntermediateInstallerVersion)' == ''">$(installerOutputPackageVersion)</MicrosoftSourceBuildIntermediateInstallerVersion>
+      <SdkFilenamePrefix>dotnet-sdk-</SdkFilenamePrefix>
+    </PropertyGroup>
+    <ItemGroup>
+      <SdkTarballItem Include="$(SourceBuiltAssetsDir)$(SdkFilenamePrefix)*$(ArchiveExtension)" />
+    </ItemGroup>
+
+    <!--
+      Extract SDK version from SDK tarball filename.
+
+      Example:
+      dotnet-sdk-9.0.100-alpha.1.24057.1-fedora.38-x64.tar.gz
+      dotnet-sdk-<SdkVersion>-<TargetRid><ArchiveExtension>
+    -->
```

Comment:
Do we need to keep this parsing logic in sync with an infrastructure that defines that file name?

Reply:
@NikolaMilosavljevic I think you missed this question